.pre-commit-config.yaml

```diff
@@ -36,6 +36,10 @@ repos:
     rev: v1.0.0
     hooks:
       - id: sphinx-lint
+  - repo: https://github.com/woodruffw/zizmor-pre-commit
+    rev: v0.9.1
+    hooks:
+      - id: zizmor
```

Comment:
It might be worth adding it as a GitHub action because I think it can catch more issues if it is passed a GITHUB_TOKEN.

Reply:
(source: https://woodruffw.github.io/zizmor/usage/#operating-modes if others are curious)

it's sort-of possible to pass ENV variables to pre-commit hooks (https://github.com/pre-commit/pre-commit/issues/758#issuecomment-404554291) though it's somewhat hacky. If we expect the hook to trigger somewhat frequently we can have it as a pre-commit hook that's disabled in CI (idk where that's documented, but https://github.com/python-trio/flake8-async/blob/f82074ad631e46daa63fa4a2b016e54dc6e762f0/.pre-commit-config.yaml#L7) and a github action - but perhaps we only need to bother with it as a github action.

Reply:
indeed you get more checks with the github action, but it looks harder to configure so I thought I'd get this one in first